tsdb/index/tsi1/partition.go

```diff
@@ -973,11 +1045,6 @@ func (p *Partition) compactToLevel(files []*IndexFile, level int, interrupt <-ch
 	default:
 	}
 
-	// Files have already been retained by caller.
-	// Ensure files are released only once.
-	var once sync.Once
-	defer once.Do(func() { IndexFiles(files).Release() })
```

Comment:
This is an unrelated change - we should ensure that even during cancellation we release the index files that were locked.